### PR TITLE
[#300] Initial code-splitting for containers

### PR DESCRIPTION
**Pull request by gabemart**
#300 

- [ ] Test plan | Unit test | Integration test
- [ ] Copyright header
- [ ] Code formatting
- [ ] Documentation

### CODE REVIEW

File: client/src/utilities/chart.js [hunk 1]
```diff
@@ -1,6 +1,11 @@
+import Immutable from 'immutable';
+
 export function getChartData(visualisation, datasets) {
   const { datasetId, spec } = visualisation;
-  const dataset = datasets[datasetId];
```
Comment (jonase) [inline client/src/utilities/chart.js]: I think the dataset should always be an Immutable object. Is this not always the case?

Comment (gabemart) [inline client/src/utilities/chart.js]: It's currently not the case when viewing a shared visualisation or dashboard - as far as I can see these functions are both broken in develop as a result

Comment (jonase) [inline client/src/utilities/chart.js]: ok, I think we should add the `fromJS` there then.

Comment (gabemart) [inline client/src/utilities/chart.js]: "There" as in the public index? Yep, that makes sense, will change.

**Comment by jonase** [comment]
@gabemart Looks good, I'll merge this.

Some follow up questions for my own understanding

* Did you follow https://webpack.github.io/docs/code-splitting.html or did you have some other reference as well?
* Do you know how and if code splitting will change for es6 modules once webpack 2 is released?

**Pull request merged**